### PR TITLE
[Fix](multi catalog)(nereids)Enable runtime filter for external table.

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

Enable runtime filter for external table.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.71 seconds
 stream load tsv:          475 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         38 seconds loaded 2358488459 Bytes, about 59 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230217041707_clickbench_pr_98574.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java [hunk 1]
```diff
@@ -143,6 +144,14 @@ public PhysicalOlapScan visitPhysicalOlapScan(PhysicalOlapScan scan, CascadesCon
         return scan;
     }
 
+    @Override
+    public PhysicalFileScan visitPhysicalFileScan(PhysicalFileScan scan, CascadesContext context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java:148]: merge it with visitPhysicalOlapScan, use visitPhysicalRelation

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterPruner.java [hunk 2]
```diff
@@ -145,6 +146,22 @@ public PhysicalOlapScan visitPhysicalOlapScan(PhysicalOlapScan olapScan, Cascade
         return olapScan;
     }
 
+    @Override
+    public PhysicalFileScan visitPhysicalFileScan(PhysicalFileScan fileScan, CascadesContext context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterPruner.java:150]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 3]
```diff
@@ -555,6 +555,11 @@ public PlanFragment visitPhysicalFileScan(PhysicalFileScan fileScan, PlanTransla
 
         Utils.execWithUncheckedException(fileScanNode::init);
         context.addScanNode(fileScanNode);
+        context.getRuntimeTranslator().ifPresent(
+                runtimeFilterGenerator -> runtimeFilterGenerator.getTargetOnScanNode(fileScan.getId()).forEach(
+                    expr -> runtimeFilterGenerator.translateRuntimeFilterTarget(expr, fileScanNode, context)
+            )
+        );
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:562]: add RF translation to all scan node translator, thanks~

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run buildall

**Pull request merged**